src/buffer_utils.cc

```diff
@@ -213,4 +213,47 @@ void write_to_debug_buffer(StringView str)
     }
 }
 
+InplaceString<23> to_string(Buffer::HistoryId id)
+{
+    if (id == Buffer::HistoryId::Invalid) {
+        InplaceString<23> res;
+        res.m_data[0] = '-';
+        res.m_length = 1;
+        return res;
+    } else {
+        return to_string(static_cast<size_t>(id));
+    }
+}
+
+String format_modification(const Buffer::Modification& modification, Quoting quoting)
```

Comment:
This does not need to be exposed, and could just be the body of `modification_as_string`